docs: Flank-scripts command overview (#1566)
Flank/flank: docs/flank-scripts/command_overview.md

# Flank-scripts command overview

### Command List

> - `release` - Group of commands for creating Flank release

Reviewer: I am wondering if most of the commands in this scope are more related to `github` than the `release`. For example, if we would have similar commands for a different provider that would be more convenient to add a new group for it and reuse the common part like `generate_release_notes`.

Author: Decided to make a separate package for `github`

    - `generate_release_notes` - Generate release notes

>     - `next_tag` - Get tag for next release

Author: fixed

    - `delete_github_release` - Delete release from Github
    - `delete_snapshot` - Delete snapshot package from artifacts repository
    - `delete_old_tag` - Delete old/unused tag from Github
    - `sync_with_maven_central` - Sync artifacts repository with Maven central
    - `make_github_release` - Make new Github release

___
    
- `coding_style` - Group of commands used for applying correct coding style
    - `apply_linter_to_git_hooks` - Apply KtLint pre-commit hook

>     - `apply_linter_to_ide` - Apply KtLint to IDE

Author: fixed

  
___
    

> - `update_dependencies` - Update repository dependencies

Author: Decided to move to a new package `github`

> - `update_dependencies` - Update repository dependencies

Author: fixed

___

- `integration_tests` - Group of commands for handling integration tests (1)
    - `process_results` - Process results of integration tests

(1) - currently there is only one command, but probably there will be more in future
___
    

> - `copy_issue_properties` - Copy properties(assignees, SP, labels) from issue to PR

Reviewer: fix - name

Author: Will be moved to new `github`

Author: fixed

    - `prepare`- Creates fresh copy of test artifacts for current working branch, basing on existing one.
    - `remove_remote` - Remove remote copy of test artifacts.
    - `resolve` - Automatically prepare local artifacts if needed.
    - `unzip` - Unpack test artifacts zip archive.
    - `upload` - Upload test artifacts zip as github release asset.
    - `zip` - Create zip archive from test artifacts directory.

___

- `firebase` - Group of commands for managing firebase integrations
    - `update_api` - Update api schema
    - `generate_client` - Generate Java Client based on api schema
    - `check_for_sdk_updates` - Check for new SDK features and create tasks for it

___
    
- `assemble` - Group of commands for assemble test application
    - `android_app` - Assemble Android test application
    - `go_artifacts` - Generate go artifacts
    - `ios_earl_grey` - Assemble iOS earl grey application
    - `ios_flank_example` - Assemble iOS flank example application
    - `ios_game_loop` - Assemble iOS game loop application
    - `ios_test_plans` - Assemble iOS test plans application

>     - `ios_all` - Assemble all iOS applications

Author: make structure assemble [ios|android] <command>

# Flank-scripts command overview

### Command List

> - `assemble` - Group of commands for assemble test application

Reviewer 2: ```suggestion
- `assemble` - Group of commands to assemble test applications
```

>     - `ios` - Subgroup of commands for managing assemble iOS apps

Reviewer 2: ```suggestion
    - `ios` - Subgroup of commands for iOS app assembly
```

        - `earl_grey` - Assemble iOS earl grey application
        - `flank_example` - Assemble iOS flank example application
        - `game_loop` - Assemble iOS game loop application
        - `test_plans` - Assemble iOS test plans application
        - `all` - Assemble all iOS applications

>     - `android` - Subgroup of command for managing assemble Android apps

Reviewer 2: ```suggestion
    - `android` - Subgroup of command for Android app assembly
```

        - `app` - Assemble Android test application
    - `go_artifacts` - Generate go artifacts
    
___

- `firebase` - Group of commands for managing firebase integrations
    - `update_api` - Update api schema
    - `generate_client` - Generate Java Client based on api schema

>     - `check_for_sdk_updates` - Check for new SDK features and create tasks for it

Reviewer 2: ```suggestion
    - `check_for_sdk_updates` - Check for new SDK features and create update tasks for it
```

   
___

- `linter` - Group of commands used for applying correct coding style
    - `apply_to_git_hooks` - Apply Linter pre-commit hook
    - `apply_to_ide` - Apply Linter to IDE
  
___

- `release` - Group of commands for creating Flank release
    - `generate_release_notes` - Generate release notes
    - `next_tag` - Get tag for next release
    - `delete_snapshot` - Delete snapshot package from artifacts repository
    - `sync_with_maven_central` - Sync artifacts repository with Maven central

___

- `test_artifacts` - Group of commands for artifacts management
    - `download` - Download test artifacts zip asset to test_artifacts directory.
    - `link` - Create symbolic link to under test_runner/src/test/kotlin/ftl/fixtures/tmp to test_artifacts/{branchName}.
    - `prepare`- Creates fresh copy of test artifacts for current working branch, basing on existing one.
    - `remove_remote` - Remove remote copy of test artifacts.
    - `resolve` - Automatically prepare local artifacts if needed.
    - `unzip` - Unpack test artifacts zip archive.
    - `upload` - Upload test artifacts zip as github release asset.
    - `zip` - Create zip archive from test artifacts directory.

  

> (1) - currently there is only one command, but probably there will be more in future

Reviewer 2: ```suggestion
(1) - denotes that there is only one command, but it may change in the future.
```
